src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -120,6 +120,12 @@ public sealed partial class QuicStream
     private long _id = -1;
     private readonly QuicStreamType _type;
 
+    /// <summary>
+    /// Provided via <see cref="StartAsync(Action{QuicStreamType}, CancellationToken)" /> from <see cref="QuicConnection" /> so that <see cref="QuicStream"/> can decrement its available stream count field.
+    /// When <see cref="HandleEventStartComplete(ref START_COMPLETE_DATA)">START_COMPLETE</see> arrives it gets invoked and unset back to <c>null</c> to not to hold any unintended reference to <see cref="QuicConnection"/>.
+    /// </summary>
+    private Action<QuicStreamType>? _decrementStreamCapacity;
+
```

Comment:
How is this better than having reference to parent QuicConection and calling an internal-only method directly?

Reply:
See these comments: https://github.com/dotnet/runtime/pull/101531#discussion_r1612094496